### PR TITLE
Render mail body in shadow dom

### PR DESCRIPTION
**Pull request by johnbotris**
see commit message

### CODE REVIEW

File: src/mail/view/MailViewerViewModel.ts [hunk 1]
```diff
@@ -113,6 +114,9 @@ export class MailViewerViewModel {
 	private readonly loadingState = new LoadingStateTracker()
 
 	private renderIsDelayed: boolean = true
+	// We render the mailbody to a shadowdom, which doesn't inherit styles from the global stylesheet
+	// So we just copy it and put it in the shadowdom
+	private styleSheet: Node = styles.getStyleSheetElement("main")
```
Comment (charlag) [inline src/mail/view/MailViewerViewModel.ts:119]: let's see if we need to continue doing it but it doesn't belong in viewModel anyway IMO

File: src/mail/view/MailViewer.ts [hunk 2]
```diff
@@ -387,17 +398,37 @@ export class MailViewer implements Component<MailViewerAttrs> {
...
+			}
 		)
 	}
 
+	/**
+	 * We render the sanitized mail body to a shadow dom to ensure that any styles don't leak into the rest of the app.
+	 * Styles in mailbodies *should* get sanitized out, but even with sanitization, there is always the possibility of bugs,
+	 * either our own or in the sanitization, so we do this as a safety precaution
+	 */
+	private renderToShadowRoot(sanitizedMailBody: string) {
```
Comment (charlag) [inline src/mail/view/MailViewer.ts:410]: you switched back to stringifying but could we do `DocumentFragment.cloneNode(true)` instead?

Comment (johnbotris) [reply src/mail/view/MailViewer.ts:410]: considered it, is it ok to do it every time?

Comment (charlag) [reply src/mail/view/MailViewer.ts:410]: Well it is cheaper than parsing it each time?

Comment (johnbotris) [reply src/mail/view/MailViewer.ts:410]: true

**Comment by charlag** [comment]
There's at least `#main-body.break-pre pre` style that is used on mobile and which we should test

**Pull request merged**